backend/controller/answers/samcrider.md

```diff
@@ -0,0 +1,41 @@
+## QUESTIONS
+
+- What is a status code? What are the most common status codes?
+
+An HTTP response status code is an indicator of whether or not an HTTP response has been successfully completed. There are five response classes: informational, successful, redirection, client error, and service error.
+
+Some common examples
+informational: 100 continue
+successful: 200 OK
+redirection: 301 Moved Permanently
+client error: 400 Bad Request, 401 Unauthorized, 404 Not Found, 418 I'm a Teapot (not common but funny)
+service error: 500 Internal Service Error, 502 Bad Gateway
+
+- What is JSON? What is XML? Why do we use them?
+
+Both of these are data representations used in data exchange between applications. JSON is the newer, more popular option.
+
+1. JSON, an open data interchange format (1), is readable by people and machines. It isn't connected to any computer languages which allows it to be a very common API output across tons of applications.
+2. XML is a markup language uses tags to differentiate between actual data and data attributes. Its rules allow it to define any data.
+
+(1): this is a format that allows applications to read and write data in it that can be used and interpreted by other applications.
+
+- What is a http request? A http response?
+
+HTTP messages are how data is exchanged between the sever and the client. HTTP requests are sent by the client and they trigger actions on the server. HTTP responses are the answer from the server, back to the client.
+
+- What is middleware?
+
+Middleware is what connects the frontend of an application to its backend. APIs are middleware. Developers use them to request and access services from software components that are outside of their immediate application.
```

Comment:
While tis is technically correct, it's also worth mentioning that middleware usually refers to "API middleware", which is often a thin layer of functions that process incoming requests and outgoing responses. 

For example, a middleware is usually responsible for authenticating a request or validating input before it reaches the controller.